clippy_lints/src/doc.rs

```diff
@@ -68,6 +68,25 @@ declare_clippy_lint! {
     "`pub unsafe fn` without `# Safety` docs"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for `fn main() { .. }` in doctests
+    ///
+    /// **Why is this bad?** The test can be shorter (and likely more readable)
+    /// if the `fn main()` is left implicit.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Examples:**
+    /// /// ```rust
```

Comment:
```suggestion
    /// ```rust
```
Same below

Reply:
On the contrary, this should be an example of a doctest within the docs.

Reply:
Docs all the way down :turtle: